nixops/resources/output.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
+# Arbitrary JSON.
+
+import os
+import nixops.util
+import nixops.resources
+
+import tempfile
+import shutil
+import subprocess
+import hashlib
+
+
+# For typing
+from nixops.deployment import Deployment
+from xml.etree.ElementTree import Element
+from nixops.nix_expr import Function
+from typing import Optional, List, Dict, Tuple
+
+class OutputDefinition(nixops.resources.ResourceDefinition):
+    """Definition of an Output."""
+
+    @classmethod
+    def get_type(cls):
+        # type: () -> (str)
+        return "output"
+
+    @classmethod
+    def get_resource_type(cls):
+        # type: () -> (str)
+        return "output"
+
+    def __init__(self, xml):
+        # type: (Element) -> None
+        nixops.resources.ResourceDefinition.__init__(self, xml)
```

Comment:
Can you get rid of the xml parsing ?
If you don't override __init__, the definition should be in a dictionary that is available through `defn.config` iirc

Reply:
Later on i detect changes between self and defn. I'm not 100% sure it is needed. I'd welcome thoughts on this.

Reply:
Yes you can still compare. The config will be available in `defn.config` inside the ResourceState.create method, so you could pull `script` from example using `defn.config['script']`
See and example of a resource implementation that does that https://github.com/NixOS/nixops/blob/master/nixops/resources/sns_topic.py

Reply:
updated.